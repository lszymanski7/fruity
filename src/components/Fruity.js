import React, { useState, useEffect } from 'react'
import Draw from './Draw'
import Footer from './Footer'
import Header from './Header'
import Modal from './Modal'
import Widget from './Widget'
import fruitsData from '../data/fruits'
import useLocalStorage from '../hooks/useLocalStorage'

const Fruity = () => {
	const [fruits, setFruits] = useLocalStorage('fruits', fruitsData)
	const [filteredFruits, setFilteredFruits] = useState(fruits.filter((fruit) => fruit.checked === true))
	const [selectedFruit, setSelectedFruit] = useState(undefined)
	const [isOpenModal, setIsOpenModal] = useState(false)

	const openModal = (fruit) => {
		setSelectedFruit(fruit)
		setIsOpenModal(true)
	}

	const closeModal = () => {
		setSelectedFruit(undefined)
		setIsOpenModal(false)
	}

	const handleDraw = () => {
		const rand = Math.floor(Math.random() * filteredFruits.length)
		openModal(filteredFruits[rand])
	}

	useEffect(() => {
		if (window.localStorage.getItem('fruits') === null) {
			setFruits(fruitsData)
		}
  		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [])

	return (
		<div>
			<Header />
			<div className="container">
				<Widget
					fruits={fruits}
					setFruits={setFruits}
					setFilteredFruits={setFilteredFruits}
				/>
				<Draw
					filteredFruits={filteredFruits}
					handleDraw={handleDraw} 
				/>
			</div>
			<Modal
				selectedFruit={selectedFruit}
				isOpenModal={isOpenModal}
				closeModal={closeModal}
			/>
			<Footer />
		</div>
	)
}

export { Fruity as default }
