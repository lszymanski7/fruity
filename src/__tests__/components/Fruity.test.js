import React from 'react'
import { render } from '@testing-library/react'
import Fruity from '../../components/Fruity'

describe('Fruity', () => {
	it('should be rendered correctly.', () => {
		const component = render(<Fruity />)
		expect(component).toMatchSnapshot()
	})
})
