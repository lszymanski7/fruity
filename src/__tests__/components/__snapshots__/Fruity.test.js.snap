// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fruity should be rendered correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="header"
        >
          <div
            class="container"
          >
            <div
              class="header-container"
            >
              <img
                alt="Fruity Logo"
                class="header__logo"
                src="logo512-outlined.png"
              />
              <h1
                class="header__title"
              >
                Fruity
              </h1>
            </div>
            <div>
              <h2
                class="header__subtitle"
              >
                Are you a fruit lover? Have you ever been unable to decide what kind of fruit you would like to eat? Don't worry about that anymore! Just click the button below!
              </h2>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="widget"
          >
            <div
              class="widget-header"
            >
              <h2>
                Fruits to select:
              </h2>
            </div>
            <div
              class="widget-container"
            >
              <div
                class="item"
              >
                <img
                  alt="Apple"
                  class="item__image"
                  id="62405bc6-aea4-4d41-9b98-58297a99c88b"
                  src="apple-color.png"
                />
                <p
                  class="item__name"
                >
                  Apple
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Banana"
                  class="item__image"
                  id="71ae9e7b-7921-4800-a832-f9086111f45b"
                  src="banana-color.png"
                />
                <p
                  class="item__name"
                >
                  Banana
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Blackberry"
                  class="item__image"
                  id="03e6fd3f-5f50-4b51-9b88-eae2d1155913"
                  src="blackberry-color.png"
                />
                <p
                  class="item__name"
                >
                  Blackberry
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Blueberries"
                  class="item__image"
                  id="2131dbb3-71d1-415b-8613-1ceaba040c90"
                  src="blueberries-color.png"
                />
                <p
                  class="item__name"
                >
                  Blueberries
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Cherries"
                  class="item__image"
                  id="a2080504-e7cf-4022-bd3b-4d42110ed514"
                  src="cherries-color.png"
                />
                <p
                  class="item__name"
                >
                  Cherries
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Coconut"
                  class="item__image"
                  id="aecd1d5e-a3e7-43e7-bf85-3d58a5444d5b"
                  src="coconut-color.png"
                />
                <p
                  class="item__name"
                >
                  Coconut
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Dragon Fruit"
                  class="item__image"
                  id="f212e4a9-bf27-4086-98cd-9c2426ef80bf"
                  src="dragon-fruit-color.png"
                />
                <p
                  class="item__name"
                >
                  Dragon Fruit
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Fig"
                  class="item__image"
                  id="11c6fb62-787f-4379-a4db-33877807be46"
                  src="fig-color.png"
                />
                <p
                  class="item__name"
                >
                  Fig
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Grapes"
                  class="item__image"
                  id="918845c1-e2ca-4cc5-934d-67c19fbec1fe"
                  src="grapes-color.png"
                />
                <p
                  class="item__name"
                >
                  Grapes
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Kiwi"
                  class="item__image"
                  id="1799f63c-d615-4303-8933-d2c4cba9257d"
                  src="kiwi-color.png"
                />
                <p
                  class="item__name"
                >
                  Kiwi
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Lemon"
                  class="item__image"
                  id="6b50a3c2-4c5c-433f-ab1f-e887397a4d76"
                  src="lemon-color.png"
                />
                <p
                  class="item__name"
                >
                  Lemon
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Lychee"
                  class="item__image"
                  id="61dda278-eafb-47c4-b403-120347eabde0"
                  src="lychee-color.png"
                />
                <p
                  class="item__name"
                >
                  Lychee
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Mango"
                  class="item__image"
                  id="3f096440-6b31-4d47-ad7e-cfb5d9ec0af2"
                  src="mango-color.png"
                />
                <p
                  class="item__name"
                >
                  Mango
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Melon"
                  class="item__image"
                  id="faaa2521-d9c3-4e28-9cbf-f7848b21587f"
                  src="melon-color.png"
                />
                <p
                  class="item__name"
                >
                  Melon
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Orange"
                  class="item__image"
                  id="798b6bed-3613-472d-89b5-3b0669ebb246"
                  src="orange-color.png"
                />
                <p
                  class="item__name"
                >
                  Orange
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Papaya"
                  class="item__image"
                  id="1c021171-431f-49e4-9577-b12534b89aae"
                  src="papaya-color.png"
                />
                <p
                  class="item__name"
                >
                  Papaya
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Peach"
                  class="item__image"
                  id="dd4d6972-18b0-4e6e-b774-800e138f3421"
                  src="peach-color.png"
                />
                <p
                  class="item__name"
                >
                  Peach
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Pear"
                  class="item__image"
                  id="e419b502-7786-455b-8ea2-c81aff093ab9"
                  src="pear-color.png"
                />
                <p
                  class="item__name"
                >
                  Pear
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Pineapple"
                  class="item__image"
                  id="07a4835c-1cd1-4f24-9754-ef95061b56b7"
                  src="pineapple-color.png"
                />
                <p
                  class="item__name"
                >
                  Pineapple
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Plum"
                  class="item__image"
                  id="24ffa800-b274-4a1f-b511-6d0dbc513521"
                  src="plum-color.png"
                />
                <p
                  class="item__name"
                >
                  Plum
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Pomegranate"
                  class="item__image"
                  id="062024ae-b117-4299-8588-c1ce6fd5c200"
                  src="pomegranate-color.png"
                />
                <p
                  class="item__name"
                >
                  Pomegranate
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Star Fruit"
                  class="item__image"
                  id="72c42012-3982-4407-9951-571144886001"
                  src="star-fruit-color.png"
                />
                <p
                  class="item__name"
                >
                  Star Fruit
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Strawberry"
                  class="item__image"
                  id="bb1ed3e5-efd8-4aa7-b6de-e313d0ea00fd"
                  src="strawberry-color.png"
                />
                <p
                  class="item__name"
                >
                  Strawberry
                </p>
              </div>
              <div
                class="item"
              >
                <img
                  alt="Watermelon"
                  class="item__image"
                  id="60b879b5-1049-4377-80cd-3687c7a1ce88"
                  src="watermelon-color.png"
                />
                <p
                  class="item__name"
                >
                  Watermelon
                </p>
              </div>
            </div>
          </div>
          <div>
            <button
              class="big-button"
            >
              Draw a fruit!
            </button>
          </div>
        </div>
        <div
          class="footer"
        >
          <p
            class="footer__attribution"
          >
            Logo created by 
            <a
              href="https://www.flaticon.com/authors/freepik"
            >
              Freepik
            </a>
             | Icons created by 
            <a
              href="https://www.flaticon.com/authors/smashicons"
            >
              Smashicons
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <div>
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="header-container"
          >
            <img
              alt="Fruity Logo"
              class="header__logo"
              src="logo512-outlined.png"
            />
            <h1
              class="header__title"
            >
              Fruity
            </h1>
          </div>
          <div>
            <h2
              class="header__subtitle"
            >
              Are you a fruit lover? Have you ever been unable to decide what kind of fruit you would like to eat? Don't worry about that anymore! Just click the button below!
            </h2>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="widget"
        >
          <div
            class="widget-header"
          >
            <h2>
              Fruits to select:
            </h2>
          </div>
          <div
            class="widget-container"
          >
            <div
              class="item"
            >
              <img
                alt="Apple"
                class="item__image"
                id="62405bc6-aea4-4d41-9b98-58297a99c88b"
                src="apple-color.png"
              />
              <p
                class="item__name"
              >
                Apple
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Banana"
                class="item__image"
                id="71ae9e7b-7921-4800-a832-f9086111f45b"
                src="banana-color.png"
              />
              <p
                class="item__name"
              >
                Banana
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Blackberry"
                class="item__image"
                id="03e6fd3f-5f50-4b51-9b88-eae2d1155913"
                src="blackberry-color.png"
              />
              <p
                class="item__name"
              >
                Blackberry
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Blueberries"
                class="item__image"
                id="2131dbb3-71d1-415b-8613-1ceaba040c90"
                src="blueberries-color.png"
              />
              <p
                class="item__name"
              >
                Blueberries
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Cherries"
                class="item__image"
                id="a2080504-e7cf-4022-bd3b-4d42110ed514"
                src="cherries-color.png"
              />
              <p
                class="item__name"
              >
                Cherries
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Coconut"
                class="item__image"
                id="aecd1d5e-a3e7-43e7-bf85-3d58a5444d5b"
                src="coconut-color.png"
              />
              <p
                class="item__name"
              >
                Coconut
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Dragon Fruit"
                class="item__image"
                id="f212e4a9-bf27-4086-98cd-9c2426ef80bf"
                src="dragon-fruit-color.png"
              />
              <p
                class="item__name"
              >
                Dragon Fruit
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Fig"
                class="item__image"
                id="11c6fb62-787f-4379-a4db-33877807be46"
                src="fig-color.png"
              />
              <p
                class="item__name"
              >
                Fig
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Grapes"
                class="item__image"
                id="918845c1-e2ca-4cc5-934d-67c19fbec1fe"
                src="grapes-color.png"
              />
              <p
                class="item__name"
              >
                Grapes
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Kiwi"
                class="item__image"
                id="1799f63c-d615-4303-8933-d2c4cba9257d"
                src="kiwi-color.png"
              />
              <p
                class="item__name"
              >
                Kiwi
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Lemon"
                class="item__image"
                id="6b50a3c2-4c5c-433f-ab1f-e887397a4d76"
                src="lemon-color.png"
              />
              <p
                class="item__name"
              >
                Lemon
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Lychee"
                class="item__image"
                id="61dda278-eafb-47c4-b403-120347eabde0"
                src="lychee-color.png"
              />
              <p
                class="item__name"
              >
                Lychee
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Mango"
                class="item__image"
                id="3f096440-6b31-4d47-ad7e-cfb5d9ec0af2"
                src="mango-color.png"
              />
              <p
                class="item__name"
              >
                Mango
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Melon"
                class="item__image"
                id="faaa2521-d9c3-4e28-9cbf-f7848b21587f"
                src="melon-color.png"
              />
              <p
                class="item__name"
              >
                Melon
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Orange"
                class="item__image"
                id="798b6bed-3613-472d-89b5-3b0669ebb246"
                src="orange-color.png"
              />
              <p
                class="item__name"
              >
                Orange
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Papaya"
                class="item__image"
                id="1c021171-431f-49e4-9577-b12534b89aae"
                src="papaya-color.png"
              />
              <p
                class="item__name"
              >
                Papaya
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Peach"
                class="item__image"
                id="dd4d6972-18b0-4e6e-b774-800e138f3421"
                src="peach-color.png"
              />
              <p
                class="item__name"
              >
                Peach
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Pear"
                class="item__image"
                id="e419b502-7786-455b-8ea2-c81aff093ab9"
                src="pear-color.png"
              />
              <p
                class="item__name"
              >
                Pear
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Pineapple"
                class="item__image"
                id="07a4835c-1cd1-4f24-9754-ef95061b56b7"
                src="pineapple-color.png"
              />
              <p
                class="item__name"
              >
                Pineapple
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Plum"
                class="item__image"
                id="24ffa800-b274-4a1f-b511-6d0dbc513521"
                src="plum-color.png"
              />
              <p
                class="item__name"
              >
                Plum
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Pomegranate"
                class="item__image"
                id="062024ae-b117-4299-8588-c1ce6fd5c200"
                src="pomegranate-color.png"
              />
              <p
                class="item__name"
              >
                Pomegranate
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Star Fruit"
                class="item__image"
                id="72c42012-3982-4407-9951-571144886001"
                src="star-fruit-color.png"
              />
              <p
                class="item__name"
              >
                Star Fruit
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Strawberry"
                class="item__image"
                id="bb1ed3e5-efd8-4aa7-b6de-e313d0ea00fd"
                src="strawberry-color.png"
              />
              <p
                class="item__name"
              >
                Strawberry
              </p>
            </div>
            <div
              class="item"
            >
              <img
                alt="Watermelon"
                class="item__image"
                id="60b879b5-1049-4377-80cd-3687c7a1ce88"
                src="watermelon-color.png"
              />
              <p
                class="item__name"
              >
                Watermelon
              </p>
            </div>
          </div>
        </div>
        <div>
          <button
            class="big-button"
          >
            Draw a fruit!
          </button>
        </div>
      </div>
      <div
        class="footer"
      >
        <p
          class="footer__attribution"
        >
          Logo created by 
          <a
            href="https://www.flaticon.com/authors/freepik"
          >
            Freepik
          </a>
           | Icons created by 
          <a
            href="https://www.flaticon.com/authors/smashicons"
          >
            Smashicons
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
